solutions/form-builder/src/app/app.component.html

```diff
@@ -0,0 +1,49 @@
+<form [formGroup]="contactForm" (ngSubmit)="addPerson()">
+
+  <div class="row">
+    <label for="firstName">First Name:</label>
+    <input type="text" name="firstName" id="firstName" [formControl]="firstName"/>
+    <div class="errors" [hidden]="firstName.valid || firstName.untouched">
+      <h4 [hidden]="!firstName.hasError('required')">
+        First name is required
+      </h4>
+      <h4 [hidden]="!firstName.hasError('minlength')">
+        First name must be at least 2 characters long
+      </h4>
```

Comment:
Because this is a list of errors I think that the appropriate html element is `<ul>` and `<li>`. Also, that's how we are doing it in the template driven forms exercise and it would be better to be consistent. Take a look at the solution of that exercise.